src/io.c

```diff
@@ -95,6 +93,9 @@ static avifResult avifIOFileReaderRead(struct avifIO * io, uint32_t readFlags, u
     }
 
     if (size > 0) {
+        if (offset > LONG_MAX) {
+            return AVIF_RESULT_IO_ERROR;
+        }
         if (reader->buffer.size < size) {
             avifRWDataRealloc(&reader->buffer, size);
         }
```

Comment:
The next line is:
        if (fseek(reader->f, (long)offset, SEEK_SET) != 0) {

So we need to make sure the (long)offset cast is safe.